pkg/cloud/services/awsnode/cni.go

```diff
@@ -239,26 +242,77 @@ func (s *Service) applyUserProvidedEnvironmentProperties(containerEnv []corev1.E
+	// EKS has a tendency to pre-install the vpc-cni automagically even if you don't specify it as an addon
+	// and looks like a kubectl apply from a script of a manifest that looks like this
+	// https://github.com/aws/amazon-vpc-cni-k8s/blob/master/config/master/aws-k8s-cni.yaml
+	// and removing these pieces will enable someone to install and alternative CNI. There is also another use
+	// case where someone would want to remove the vpc-cni and reinstall it via the helm chart located here
+	// https://github.com/aws/amazon-vpc-cni-k8s/tree/master/charts/aws-vpc-cni meaning we need to account for
+	// managed-by: Helm label, or we will delete the helm chart resources every reconcile loop. EKS does make
+	// a CRD for eniconfigs but the default env var on the vpc-cni pod is ENABLE_POD_ENI=false. We will make an
+	// assumption no CRs are ever created and leave the CRD to reduce complexity of this operation.
+
+	s.scope.Info("Ensuring all resources for AWS VPC CNI in cluster are deleted", "cluster-name", s.scope.Name(), "cluster-namespace", s.scope.Namespace())
+
+	s.scope.Info("Trying to delete AWS VPC CNI DaemonSet", "cluster-name", s.scope.Name(), "cluster-namespace", s.scope.Namespace())
+	if err := s.deleteResource(ctx, remoteClient, types.NamespacedName{
+		Namespace: awsNodeNamespace,
+		Name:      awsNodeName,
+	}, &appsv1.DaemonSet{}); err != nil {
+		return err
+	}
 
-	ds := &appsv1.DaemonSet{}
-	if err := remoteClient.Get(ctx, types.NamespacedName{Namespace: awsNodeNamespace, Name: awsNodeName}, ds); err != nil {
-		if apierrors.IsNotFound(err) {
-			s.scope.V(2).Info("The aws-node DaemonSet is not found, not action")
-			return nil
-		}
-		return fmt.Errorf("getting aws-node daemonset: %w", err)
+	s.scope.Info("Trying to delete AWS VPC CNI ServiceAccount", "cluster-name", s.scope.Name(), "cluster-namespace", s.scope.Namespace())
+	if err := s.deleteResource(ctx, remoteClient, types.NamespacedName{
+		Namespace: awsNodeNamespace,
+		Name:      awsNodeName,
+	}, &corev1.ServiceAccount{}); err != nil {
+		return err
 	}
 
-	s.scope.V(2).Info("The aws-node DaemonSet found, deleting")
-	if err := remoteClient.Delete(ctx, ds, &client.DeleteOptions{}); err != nil {
-		if apierrors.IsNotFound(err) {
-			s.scope.V(2).Info("The aws-node DaemonSet is not found, not deleted")
-			return nil
-		}
-		return fmt.Errorf("deleting aws-node DaemonSet: %w", err)
+	s.scope.Info("Trying to delete AWS VPC CNI ClusterRoleBinding", "cluster-name", s.scope.Name(), "cluster-namespace", s.scope.Namespace())
+	if err := s.deleteResource(ctx, remoteClient, types.NamespacedName{
+		Namespace: string(meta.RESTScopeNameRoot),
+		Name:      awsNodeName,
+	}, &rbacv1.ClusterRoleBinding{}); err != nil {
+		return err
 	}
+
+	s.scope.Info("Trying to delete AWS VPC CNI ClusterRole", "cluster-name", s.scope.Name(), "cluster-namespace", s.scope.Namespace())
```

Comment:
You aren't deleting the DaemonSet. Right?

Reply:
yes, it was always doing that and happens at line 261 now